build/hostlogswindows/installer/livenessprobe/livenessprobe.cpp

```diff
@@ -7,37 +7,87 @@
 #endif
 
 #include <Windows.h>
-#include <tchar.h>
+#include <tlhelp32.h>
+#include <iostream>
 
 #define SUCCESS 0x00000000
-#define FILESYSTEM_WATCHER_FILE_EXISTS 0x00000001
+#define NO_MONAGENT_LAUNCHER_PROCES 0x00000001
+#define FILESYSTEM_WATCHER_FILE_EXISTS 0x00000002
 #define UNEXPECTED_ERROR 0xFFFFFFFF
 
 /*
-  check if the file exists
+*  Function to check if the given file exists
 */
-bool IsFileExists(const wchar_t *const fileName)
+bool IsFileExists(const wchar_t* const fileName)
 {
     DWORD dwAttrib = GetFileAttributes(fileName);
     return dwAttrib != INVALID_FILE_SIZE;
 }
 
+/*
+* Function to check if a process is running based on its full path
+*/ 
+bool IsMonAgentRunning(const wchar_t* executableFullPath) {
+    PROCESSENTRY32 processEntry{};  // struct to hold process details
+    processEntry.dwSize = sizeof(PROCESSENTRY32);  // set size of struct
+
+    // Handle to snapshot of all processes currently running on the system
+    HANDLE processSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+
+    // Traverse through the list of processes in the snapshot
+    if (Process32First(processSnapshot, &processEntry)) {
+        do {
+            // For each process, capture a snapshot of all modules
+            HANDLE moduleSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPMODULE, processEntry.th32ProcessID);
+
+            MODULEENTRY32 moduleEntry{}; // struct to hold details of a module (exe) associated with a process
+            moduleEntry.dwSize = sizeof(MODULEENTRY32); // set size of struct
+
+            // If the module (exe) path matches with the given path, process is running
+            if (Module32First(moduleSnapshot, &moduleEntry) && !_wcsicmp(moduleEntry.szExePath, executableFullPath)) {
+                CloseHandle(moduleSnapshot);
+                CloseHandle(processSnapshot);
+                return true;
+            }
+
+            CloseHandle(moduleSnapshot);
+        } while (Process32Next(processSnapshot, &processEntry));
+    }
+
+    CloseHandle(processSnapshot);
+    return false;
+}
+
 /**
- <exe> <filesystemwatcherfilepath>
+ <exe> <monAgentLauncherExe> <filesystemwatcherfilepath>
 **/
-int _tmain(int argc, wchar_t *argv[])
+int wmain(int argc, wchar_t* argv[]) 
 {
-    if (argc < 2)
+    try 
     {
-        wprintf_s(L"ERROR:unexpected number arguments and expected is 5");
-        return UNEXPECTED_ERROR;
-    }
+        if (argc < 3) 
+        {  
+            wprintf_s(L"ERROR:unexpected number arguments and expected is 5");
```

Comment:
Update error message. We expect 2 args

Reply:
C++ always puts the livenessprobe.exe at argv[0]

https://github.com/microsoft/Docker-Provider/blob/290f4acbecdf18742781f539c3fb400eafd8c183/build/windows/installer/livenessprobe/livenessprobe.cpp#L106

https://github.com/microsoft/Docker-Provider/blob/290f4acbecdf18742781f539c3fb400eafd8c183/build/windows/installer/livenessprobe/livenessprobe.cpp#L110

https://github.com/microsoft/Docker-Provider/blob/290f4acbecdf18742781f539c3fb400eafd8c183/kubernetes/ama-logs.yaml#L997-L1006